cylc/flow/task_pool.py

```diff
@@ -461,7 +462,12 @@ def load_db_task_pool_for_restart(self, row_idx, row):
                     TASK_STATUS_SUCCEEDED
             ):
                 # update the task proxy with platform
-                itask.platform = get_platform(platform_name)
+                # If we get a failure from the platform selection function
```

Comment:
Again a case where I'm not convinced platform name could ever be a platform group. I think it's safe to say that any platform which has become inaccessible/been removed from the config (I think that the latter is much more likely here) that the task's submission has failed. This then triggers whatever retry logic the user wants.

Reply:
This should hopefully be a really niche error only really possible due to global config changes since restart. The most likely case for this error is probably a really old workflow being restarted after a global config change.

E.G. we currently have two Cylc platforms representing a single real platform just with different configuration for back-compat reasons. One day we'll remove the back-compat platform, this error could then ensue, yikes!

Submit-failed is definitely wrong because the last recorded status was submitted or greater.
* In the case of succeeded and failed tasks the platform shouldn't really matter any more but I guess it might still be used by some command or another?! Or possibly recorded back to the DB with a task pool dump!?
* For submitted / running tasks it's more of a problem as Cylc will try to poll them.

Perhaps the correct behaviour here is to raise an exception :grin:! We could potentially collect any `PlatformLookupError`s, list the tasks in the error with a suggestion to hack the DB to allow restart to happen.